### PR TITLE
2023-05 logic tweaks

### PR DESCRIPTION
**Pull request by kbranch**
This has two logic tweaks:
- Allow jesus jumping to Manbo's cave without the angler key in hell logic. I did this by splitting the watery area to the left of the D4 entrance into its own location and connecting that to the shore. Note that I removed Manbo's entrance requirements because it seems redundant now, but please double check me there.
- Make the requirements for Richard's owl the same as the slime key dig (minus the shovel). The powder upgrade wasn't originally required for the owl, but it was for the dig.

### CODE REVIEW

**Comment by kbranch** [comment]
Just pushed a tweak to allow escaping from Richard's maze without the powder upgrade

**Comment by kbranch** [comment]
One more, putting featherless fisher back into hard logic. It's easy but not well known. I've seen at least two people assume it's a logic bug so far.

File: logic/overworld.py [hunk 1]
```diff
@@ -226,9 +223,9 @@ def __init__(self, options, world_setup, r):
         richard_maze = Location()
         self._addEntrance("richard_house", ukuku_prairie, richard_house, None)
         self._addEntrance("richard_maze", richard_maze, richard_cave, None)
-        if options.owlstatues == "both" or options.owlstatues == "overworld":
-            Location().add(OwlStatue(0x0C6)).connect(richard_maze, r.bush)
-        Location().add(SlimeKey()).connect(richard_maze, AND(OR(SWORD, AND(MAGIC_POWDER, MAX_POWDER_UPGRADE), MAGIC_ROD, POWER_BRACELET, BOOMERANG), SHOVEL))
+        slime_key_area = Location().connect(richard_maze, OR(SWORD, AND(MAGIC_POWDER, MAX_POWDER_UPGRADE), MAGIC_ROD, POWER_BRACELET, BOOMERANG))
+        Location().add(OwlStatue(0x0C6)).connect(slime_key_area, options.owlstatues in ('both', 'overworld'))
```
Comment (daid) [inline logic/overworld.py:227]: Any particular reason this was changed from an `if` into a `true/false` connection? It doesn't match the other owl statues this way, and might confuse the itempool.

Comment (kbranch) [reply logic/overworld.py]: Nope, it was just on a whim for aesthetics. But you're right, even then it doesn't fit the rest of the owls.

Just so I understand better, is the item pool concern that there will be a mismatch between item count and places to put the items? I did briefly try generation without owls with that in mind, but didn't run into a problem. I assumed it was safely disconnected from the graph when the condition was false.

**Comment by daid** [comment]
I think there are a bunch of conflicts now because I finally merged the other pull request on the logic.

**Comment by kbranch** [comment]
Yep, makes sense - I think the merge looks clean now.

**Pull request merged**